httpclient-backend/zio/src/main/scala/sttp/client/httpclient/zio/HttpClientZioBackend.scala

```diff
@@ -1,62 +1,147 @@
 package sttp.client.httpclient.zio
 
+import java.io.InputStream
+import java.net.http.HttpRequest.BodyPublishers
 import java.net.http.{HttpClient, HttpRequest}
+import java.nio.ByteBuffer
 
+import org.reactivestreams.FlowAdapters
 import sttp.client.httpclient.HttpClientBackend.EncodingHandler
 import sttp.client.httpclient.{HttpClientAsyncBackend, HttpClientBackend}
+import sttp.client.httpclient.zio.HttpClientZioBackend.BlockingTask
 import sttp.client.impl.zio.RIOMonadAsyncError
 import sttp.client.{FollowRedirectsBackend, SttpBackend, SttpBackendOptions}
 import zio._
+import zio.blocking.Blocking
+import zio.interop.reactivestreams._
+import zio.stream.{Stream, ZStream}
+
+import scala.util.{Success, Try}
 
 class HttpClientZioBackend private (
     client: HttpClient,
     closeClient: Boolean,
     customizeRequest: HttpRequest => HttpRequest,
-    customEncodingHandler: EncodingHandler
-) extends HttpClientAsyncBackend[Task, Nothing](
+    customEncodingHandler: EncodingHandler,
+    chunkSize: Int
+) extends HttpClientAsyncBackend[BlockingTask, ZStream[Blocking, Throwable, Byte]](
       client,
-      new RIOMonadAsyncError,
+      new RIOMonadAsyncError[Blocking],
       closeClient,
       customizeRequest,
       customEncodingHandler
-    )
+    ) {
+
+  override def streamToRequestBody(
+      stream: ZStream[Blocking, Throwable, Byte]
+  ): BlockingTask[HttpRequest.BodyPublisher] = {
+    val publisher = stream.mapChunks(byteChunk => Chunk(ByteBuffer.wrap(byteChunk.toArray))).toPublisher
+    publisher.map { pub =>
+      BodyPublishers.fromPublisher(FlowAdapters.toFlowPublisher(pub))
+    }
+  }
+
+  override def responseBodyToStream(responseBody: InputStream): Try[ZStream[Blocking, Throwable, Byte]] = {
+    Success(Stream.fromInputStreamEffect(ZIO.succeed(responseBody), chunkSize))
+  }
+}
 
 object HttpClientZioBackend {
+  type BlockingTask[A] = ZIO[Blocking, Throwable, A]
```

Comment:
Do we need blocking in the effect type? Or only for streams?

Reply:
Yes, because we use InputStream for reading response body for HttpClient backend.

Reply:
Ah yes, that's not only for streams :)